bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/aws/http/AuthenticationHelper.java

```diff
@@ -0,0 +1,436 @@
+            + "EF9519B3CD3A431B302B0A6DF25F14374FE1356D6D51C245"//
+            + "E485B576625E7EC6F44C42E9A637ED6B0BFF5CB6F406B7ED"//
+            + "EE386BFB5A899FA5AE9F24117C4B1FE649286651ECE45B3D"//
+            + "C2007CB8A163BF0598DA48361C55D39A69163FA8FD24CF5F"//
+            + "83655D23DCA3AD961C62F356208552BB9ED529077096966D"//
+            + "670C354E4ABC9804F1746C08CA18217C32905E462E36CE3B"//
+            + "E39E772C180E86039B2783A2EC07A28FB5C55DF06F4C52C9"//
+            + "DE2BCBF6955817183995497CEA956AE515D2261898FA0510"//
+            + "15728E5A8AAAC42DAD33170D04507A33A85521ABDF1CBA64"//
+            + "ECFB850458DBEF0A8AEA71575D060C7DB3970F85A6E1E4C7"//
+            + "ABF5AE8CDB0933D71E8C94E04A25619DCEE3D2261AD2EE6B"//
+            + "F12FFA06D98A0864D87602733EC86A64521F2B18177B200C"//
+            + "BBE117577A615D6C770988C0BAD946E208E24FA074E5AB31"//
+            + "43DB5BFCE0FD108E4B82D120A93AD2CAFFFFFFFFFFFFFFFF";
+    private static final BigInteger N = new BigInteger(HEX_N, 16);
+    private static final BigInteger g = BigInteger.valueOf(2);
+    private static final BigInteger k;
+    private static final int EPHEMERAL_KEY_LENGTH = 1024;
+    private static final int DERIVED_KEY_SIZE = 16;
+    private static final String DERIVED_KEY_INFO = "Caldera Derived Key";
+    private static final ThreadLocal<MessageDigest> THREAD_MESSAGE_DIGEST = ThreadLocal.withInitial(() -> {
+        try {
+            return MessageDigest.getInstance("SHA-256");
+        } catch (NoSuchAlgorithmException e) {
+            throw new SecurityException("Exception in authentication", e);
+        }
+    });
+    private static final SecureRandom SECURE_RANDOM;
+    private static final String ALGORITHM = "HmacSHA256";
+
+    static {
+        try {
+            SECURE_RANDOM = SecureRandom.getInstance("SHA1PRNG");
+
+            MessageDigest messageDigest = THREAD_MESSAGE_DIGEST.get();
+            messageDigest.reset();
+            messageDigest.update(N.toByteArray());
+            byte[] digest = messageDigest.digest(g.toByteArray());
+            k = new BigInteger(1, digest);
+        } catch (NoSuchAlgorithmException e) {
+            throw new SecurityException(e.getMessage(), e);
+        }
+    }
+
+    private BigInteger a;
+    private BigInteger A;
+    private final String userPoolID;
+    private final String clientId;
+    private final String region;
+
+    AuthenticationHelper(String userPoolID, String clientid, String region) {
+        do {
+            a = new BigInteger(EPHEMERAL_KEY_LENGTH, SECURE_RANDOM).mod(N);
+            A = g.modPow(a, N);
+        } while (A.mod(N).equals(BigInteger.ZERO));
+
+        this.userPoolID = userPoolID;
+        this.clientId = clientid;
+        this.region = region;
+    }
+
+    private byte[] getPasswordAuthenticationKey(String userId, byte[] userPassword, BigInteger B, BigInteger salt)
+            throws ShortBufferException, NoSuchAlgorithmException, InvalidKeyException {
```

Comment:
`THREAD_MESSAGE_DIGEST` throws `SecurityException` and that is not in the list of thrown excpetions and there is no try/catch block.

Reply:
this lines throws  NoSuchAlgorithmException, InvalidKeyException, ShortBufferException

```java
hkdf.init(S.toByteArray(), u.toByteArray());
return hkdf.deriveKey(DERIVED_KEY_INFO, DERIVED_KEY_SIZE);
```

Reply:
Yep, so very good that those Exceptions are listed, my point is the  `SecurityException` should be there too

Reply:
OK, I've added it